Linker issue with gcc 10

When building with gcc 10.2.1 there are linker errors like this:

    /usr/lib64/gcc/x86_64-suse-linux/10/../../../../x86_64-suse-linux/bin/ld: log.c.o (symbol from plugin): in function `LogModuleName':
    (.text+0x0): multiple definition of `gLogFile'; json_utils.c.o (symbol from plugin):(.text+0x0): first defined here
    
    (.text+0x0): multiple definition of `gLogFile'; grafx.c.o (symbol from plugin):(.text+0x0): first defined here
    /usr/lib64/gcc/x86_64-suse-linux/10/../../../../x86_64-suse-linux/bin/ld: nine_slice.c.o (symbol from plugin): in function `Draw9Slice':
    (.text+0x0): multiple definition of `gLogFile'; grafx.c.o (symbol from plugin):(.text+0x0): first defined here
    
    /usr/lib64/gcc/x86_64-suse-linux/10/../../../../x86_64-suse-linux/bin/ld: config_json.c.o (symbol from plugin): in function `gLogFile':
    (.text+0x0): multiple definition of `gLogFile'; config_io.c.o (symbol from plugin):(.text+0x0): first defined here
    /usr/lib64/gcc/x86_64-suse-linux/10/../../../../x86_64-suse-linux/bin/ld: json_utils.c.o (symbol from plugin): in function `AddIntPair':
    (.text+0x0): multiple definition of `gLogFile'; config_io.c.o (symbol from plugin):(.text+0x0): first defined here
    /usr/lib64/gcc/x86_64-suse-linux/10/../../../../x86_64-suse-linux/bin/ld: log.c.o (symbol from plugin): in function `LogModuleName':
    (.text+0x0): multiple definition of `gLogFile'; config_io.c.o (symbol from plugin):(.text+0x0): first defined here

Looks like the problem is the definition of gLogFile in log.h instead of log.c.
This patch works for me:
[cdogs-gLog.patch.txt](https://github.com/cxong/cdogs-sdl/files/5305050/cdogs-gLog.patch.txt)

--- a/src/cdogs/log.c
+++ b/src/cdogs/log.c
@@ -32,6 +32,7 @@
 #include "rlutil/rlutil.h"
 #include "utils.h"
 
+FILE *gLogFile;
 
 typedef struct
 {

--- a/src/cdogs/log.h
+++ b/src/cdogs/log.h
@@ -60,7 +60,7 @@ void LogModuleSetLevel(const LogModule m, const LogLevel l);
 const char *LogLevelName(const LogLevel l);
 LogLevel StrLogLevel(const char *s);
 
-FILE *gLogFile;
+extern FILE *gLogFile;
 void LogInit(void);
 void LogOpenFile(const char *filename);
 void LogTerminate(void);

